### PR TITLE
Add support for ArbOS

### PR DESCRIPTION
**Pull request by TheMysteriousX**
This adds basic support for PeakFlow appliances. They don't really expose anything interesting by SNMP beyond what a generic device has.

I've referred to everything as 'arbos', but I don't know if the behaviour here covers APS/TMS devices too.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by TheMysteriousX** [comment]
I don't understand the test failure - I've not done anything that should affect 3com devices.

File: tests/snmpsim/arbos.snmprec [hunk 1]
```diff
@@ -0,0 +1,630 @@
+.1.3.6.1.2.1.1.1.0 = STRING: Peakflow SP 8.3 (build HIU6)  System Board Model: S2600CO Serial Number: CG22541072COI
```
Comment (murrant) [inline tests/snmpsim/arbos.snmprec]: This is an snmpwalk, not an snmprec file.  The format is wrong.

That is what is causing tests to fail.

See https://docs.librenms.org/Developing/os/Test-Units/

**Comment by murrant** [comment]
You can also remove the empty file includes/definitions/discovery/arbos.yaml (unless you intend to add some sensors there)

**Comment by TheMysteriousX** [comment]
thanks for the pointer on the snmprec file - not sure how I ended up with a walk capture there.

I'll remove other yaml file - there aren't any sensors exposed that I can find.

One thing I'll try and add is load average, which is exposed through their own MIB. Raw CPU information isn't exposed.

**Comment by TheMysteriousX** [comment]
I can think of three ways to do this. 

* I can tack it on to the end of ```includes/polling/ucd-mib.inc.php``` (logical, as it is the same information, just exposed at a different oid)
* I could add it into ```includes/polling/os/arbos.inc.php``` and still place it into the ```ucd_load``` rrd (may conflict with ```ucd.inc.php``` if Arbor start using standard oids)
* I can add an additional (duplicate) load average definition.

Any of these preferred (or something else?).

**Comment by laf** [comment]
So this is a bit tricky as ucd-mib does a lot more than just load average but I don't think I'm personally a fan of tacking it on to the end of the os polling.

Maybe be worth adding a directory `includes/polling/ucd-mib/` and have the new OS file in that.

Then before where the load snmp_get is called do an is_file check for `includes/polling/ucd-mib/$os.inc.php`, if it exists load it, do you call for data in there and populate the variables needed for the rrd file. Then do `} else {` and the normal snmp call - make sense?

@murrant - Thoughts? Again this is better when it gets converted to OOP but seems an ok way to deal with this and any future devices that require similar.

**Comment by murrant** [comment]
@laf In the processor rewrite branch, we always check for ucd mib if no other processors have been found.

**Comment by TheMysteriousX** [comment]
Sounds like the best thing to do would be to wait until #8066 is merged before I add load averages in.

I've cleaned up this request, so I'm happy to submit for review, and I'll open another request with load at a later date.

**Comment by laf** [comment]
> @laf In the processor rewrite branch, we always check for ucd mib if no other processors have been found.

Talking about load here rather than processors stuff so not related.

**Pull request merged**